### PR TITLE
[irods/irods7734] iadmin failures now exit non-zero (main)

### PR DESCRIPTION
**Pull request by FifthPotato**
Fixes all the test cases listed in the ticket (they now return 4).
Pending tests, but I think those need to be included in a PR to the main server(?)

### CODE REVIEW

**Comment by korydraughn** [comment]
Yes, the tests go in the irods/irods repo.

File: src/iadmin.cpp [hunk 1]
```diff
@@ -2001,10 +2001,13 @@ main( int argc, char **argv ) {
...
         if ( firstTime ) {
-            if ( status == 0 ||
-                    status == -2 ) {
+            if (0 == status) {
                 keepGoing = false;
             }
+            if (-2 == status) {
+                keepGoing = false;
+                lastCommandStatus = -2;
+            }
```
Comment (korydraughn) [inline src/iadmin.cpp:2010]: Please add words explaining why -2 is special. If you know what the distinguishing factor is between the different error codes, put it in a comment as well.

Can even add words to the commit body.

**Comment by korydraughn** [comment]
Once you open the irods/irods PR for testing this, link this PR to it. You can update the original comment of this PR to include a statement like the following:
```
Depends on irods/irods#<PR_number>.
```

**Review by alanking — COMMENTED** [review]
I think we can go ahead and squash this one. Will wait to approve / # / merge until after irods/irods#8030 is ready

**Comment by trel** [comment]
agreed

**Comment by FifthPotato** [comment]
Squished.

**Comment by korydraughn** [comment]
Let us know once the full test run passes.

**Comment by FifthPotato** [comment]
All core and unit tests passed.

**Review by korydraughn — APPROVED** [review]
Pound it.

**Comment by FifthPotato** [comment]
Octothorpe'd

**Comment by korydraughn** [comment]
Please cherry pick to 4-3-stable

**Pull request merged**